src/geometry/ArborX_DetailsAlgorithms.hpp

```diff
@@ -397,6 +412,45 @@ struct intersects<PointTag, SphereTag, Point, Sphere>
   }
 };
 
+template <typename Point, typename Triangle>
+struct intersects<PointTag, TriangleTag, Point, Triangle>
+{
+  KOKKOS_FUNCTION static bool apply(Point const &point,
```

Comment:
Any particular reason why this is not constexpr?
Same comment for expand.

Reply:
`expand` changes the objects, with the first argument being non-const.

Reply:
OK we can fix later. This will involve more changes.